perception/lidar_centerpoint/lib/preprocess/pointcloud_densification.cpp

```diff
@@ -84,12 +85,24 @@ bool PointCloudDensification::enqueuePointCloud(
 }
 
 void PointCloudDensification::enqueue(
-  const sensor_msgs::msg::PointCloud2 & msg, const Eigen::Affine3f & affine_world2current)
+  const sensor_msgs::msg::PointCloud2 & msg, const Eigen::Affine3f & affine_world2current,
+  cudaStream_t stream)
 {
   affine_world2current_ = affine_world2current;
   current_timestamp_ = rclcpp::Time(msg.header.stamp).seconds();
-  PointCloudWithTransform pointcloud = {msg, affine_world2current.inverse()};
-  pointcloud_cache_.push_front(pointcloud);
+
+  assert(sizeof(uint8_t) * msg.width * msg.height * msg.point_step % sizeof(float) == 0);
+  auto points_d = cuda::make_unique<float[]>(
+    sizeof(uint8_t) * msg.width * msg.height * msg.point_step / sizeof(float));
+  CHECK_CUDA_ERROR(cudaMemcpyAsync(
+    points_d.get(), msg.data.data(), sizeof(uint8_t) * msg.width * msg.height * msg.point_step,
+    cudaMemcpyHostToDevice, stream));
```

Comment:
(let me know if my understanding is wrong :pray: )

My understanding is that this Memcpy from host to device runs each time when the node receives a new pointcloud and hold them as long as the queue size is less than or equal to `param_.pointcloud_cache_size()`.

Here's my question: Is it safe to Memcpy a data, holds it for 0.1[sec] x n seconds, and then use it in `generateSweepPoints`? What would happen if any other process which is also using the same GPU overwrites the copied memory on the GPU device?

Reply:
I am sorry, I do not really understand the question.
If process A allocates N1 bytes in address A1, that address is only known to A.
If processes B,C... only work inside the memory allocates by them, they should never access A1.
cudamalloc of course handles things so that the space [A1, A1 + N1] is reserved by A.

If that was not the question we can talk any day in the office

Reply:
Understood, that answers my question. Thanks!